src/intrinsic.rs

```diff
@@ -286,6 +286,14 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
                 this.write_scalar(result_ptr, dest)?;
             }
 
+            "panic_if_uninhabited" => {
+                let ty = substs.type_at(0);
+                let layout = this.layout_of(ty)?;
+                if layout.abi.is_uninhabited() {
+                    return err!(Intrinsic(format!("Trying to instantiate uninhabited type {}", ty)))
```

Comment:
Shouldn't this be a [`Panic`](https://github.com/rust-lang/rust/blob/0194bed71070905e102a82e5f4643ad6ced62976/src/librustc/mir/interpret/error.rs#L278) error? I'm assuming line and file info would be the info from the call site.

Reply:
Technically, yes. I didn't feel that was worth the effort though... we can't catch panics anyway.